### DISCUSSION TITLE
I'm still working on the project

### DISCUSSION THREAD

**Discussion by metafates** [metafates/mangal#175] [Announcements]
Thank you, everyone, for the 700 stars. I'm very grateful for the recognition this project got. Especially for the https://github.com/oae/kaizoku project - it's incredible and makes me proud of what people can accomplish with Mangal.

Lately, for the last couple of months, I didn't fix any existing bugs, nor responded to the new issues. I'll be honest - I lost interest in mangal. I was focused on the University studies, other projects, personal life. But, lately, I've got the spark again and decided to make it different this time. Make it better and more thoughtful.

Soon I will release an improved, rewritten from scratch and redesigned mangal engine (as a Go module) that would make it possible to write custom mangal clients, such as cli, tui, gui, mobile, with ease. It already works well, so I'm just polishing everything and writing documentation.

Later, I will rewrite Mangal too and new major version 5 will utilize this engine. I'll try to make this new TUI client look simillar to as it was before, but of course, it will be different in some ways.

Thank you!

**Update:** Mangal engine is now available here https://github.com/mangalorg/libmangal along with https://github.com/mangalorg/luaprovider 

Note, that it's still highly experimental and may change at any time.

![_16fdf945-f7ac-454c-b6e5-e2092613261e-min](https://github.com/metafates/mangal/assets/62389790/233eb7ca-cf5d-498f-9ebd-6b84723bacca)

**Comment by justchokingaround** [comment]
It's important to take time off sometimes and take breaks when you need them. Wish you all the best with the rewrite, mangal is amazing and I hope it becomes even more amazing in the future, keep up the good work!

**Comment by OmiChrome** [comment]
i love how metafates has evolved with mangal 
i been following you since mangal was just born 
and you have grown a LOt

its  okay to loss interest in something its human nature

just looking for the good work you have planed ! :)
